x/genutil/client/cli/module_hash_by_height.go

```diff
@@ -0,0 +1,107 @@
+package cli
+
+import (
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/spf13/cobra"
+
+	"cosmossdk.io/store/rootmulti"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/server"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+)
+
+// ModuleHashByHeightQuery retrieves the module hashes at a given height.
+func ModuleHashByHeightQuery[T servertypes.Application](appCreator servertypes.AppCreator[T]) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "module-hash-by-height [height]",
+		Short: "Get module hashes at a given height",
+		Long: `Get module hashes at a given height. This command is useful for debugging and verifying the state of the application at a given height. Daemon should not be running when calling this command.
+Example:
+	osmosisd module-hash-by-height 16841115,
+`,
+		Args: cobra.ExactArgs(1), // Ensure exactly one argument is provided
+		RunE: func(cmd *cobra.Command, args []string) error {
+			heightToRetrieveString := args[0]
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+
+			height, err := strconv.ParseInt(heightToRetrieveString, 10, 64)
+			if err != nil {
+				return err
+			}
+
+			commitInfoForHeight, err := getModuleHashesAtHeight(serverCtx, appCreator, height)
+			if err != nil {
+				return err
+			}
+
+			// Print the CommitInfo to the console.
+			fmt.Println(commitInfoForHeight.String())
```

Comment:
nit: allow json

Reply:
Added json functionality here, I believe it was done to standard but would appreciate an ack https://github.com/cosmos/cosmos-sdk/pull/20779/commits/8e2f8bb1240f0a7d148e12f6707f37d3dad6fc7e